### ISSUE TITLE
Local edge functions\ does not working when WebAssembly.instantiate() is called inside.

### ISSUE THREAD

**Issue by haschu** [supabase/supabase#13502]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Hey!

My company is about to switch to supabase for our upcoming projects. So far everything has gone smoothly, but now we're experiencing a strange issue:

When `WebAssembly.instantiate` or `WebAssembly.instantiateStreaming` is called inside an edge function (served locally), it only works for the very first request. After that, a 502 error response ("an invalid response was received from the upstream server") is sent for each request.

However, when the same function is deployed, it just works without any problems. 

`supabase functions serve --debug` doesn't log any errors.

What's also strange is, that the complete code inside that function works as expected and all console.logs regarding that wasm are printed to the console - no errors. We're just not receiving the response.

I'm glad for any help! We are really stuck here :/

## To Reproduce

Here's an example function:

```ts
import { serve } from 'https://deno.land/std@0.168.0/http/server.ts';
import { init } from 'https://esm.sh/@dqbd/tiktoken/lite/init';

serve(async req => {
  await init(async imports => {
    return WebAssembly.instantiateStreaming(
      fetch('https://esm.sh/@dqbd/tiktoken@1.0.3/lite/tiktoken_bg.wasm'),
      imports,
    );
  });

  return new Response(
    JSON.stringify({
      hello: 'world',
    }),
    {
      headers: { 'Content-Type': 'application/json' },
    },
  );
});
```

- Run `supabase functions serve`
- Make a request to that functions: very first request succeeds. After that, "An invalid response was received from the upstream server" responses will be sent.
- Deploy that function: everything works as expected

## Expected behavior

The egde function should work locally.

## System information

- OS: macOS (M1)
- Version of Node.js: v18.14.2

**Comment by laktek** [comment]
@haschu This issue should be fixed in the latest release of Supabase CLI (1.50+). Let us know if you still continue to experience the issue.

**Comment by haschu** [comment]
@laktek Works flawlessly. Thanks a lot for the fast and supa support! 😁
